flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/reader/KafkaPartitionSplitReader.java

```diff
@@ -147,33 +133,17 @@ public KafkaPartitionSplitReader(
                             recordsBySplits);
                     break;
                 }
-                // Add the record to the partition collector.
-                try {
-                    deserializationSchema.deserialize(consumerRecord, collector);
-                    collector
-                            .getRecords()
-                            .forEach(
-                                    r ->
-                                            recordsForSplit.add(
-                                                    new Tuple3<>(
-                                                            r,
-                                                            consumerRecord.offset(),
-                                                            consumerRecord.timestamp())));
-                    // Finish the split because there might not be any message after this point.
-                    // Keep polling
-                    // will just block forever.
-                    if (consumerRecord.offset() == stoppingOffset - 1) {
-                        finishSplitAtRecord(
-                                tp,
-                                stoppingOffset,
-                                consumerRecord.offset(),
-                                finishedPartitions,
-                                recordsBySplits);
-                    }
-                } catch (Exception e) {
-                    throw new IOException("Failed to deserialize consumer record due to", e);
-                } finally {
-                    collector.reset();
+                recordsForSplit.add(consumerRecord);
```

Comment:
Given that the deserialization is no longer performed in the split fetcher. It seems that we don't have to iterate over all the records here any more. Instead, the `KafkaPartitionSplitRecords` can be changed to a wrapper around `ConsumerRecords`.

Reply:
Thanks for the review @becketqin ! Fixed in 678071d